Assets/Plugins/Editor/Uplift/UpliftManager.cs

```diff
@@ -79,13 +79,78 @@ public enum InstallStrategy {
             UPDATE_LOCKFILE
         }
 
+        public struct DependencyState
+        {
+            public DependencyDefinition definition;
+            public InstalledPackage installed;
+            public PackageRepo bestMatch, latest;
+            public bool transitive;
+        }
+        
         public void InstallDependencies(InstallStrategy strategy = InstallStrategy.UPDATE_LOCKFILE, bool refresh = false)
         {
             if (refresh) UpliftManager.ResetInstances();
             PackageRepo[] targets = GetTargets(GetDependencySolver(), strategy);
             InstallPackages(targets);
         }
 
+        public DependencyState[] GetDependenciesState(bool refresh = false)
```

Comment:
Do we want to add the install strategy argument as in the above method's API? Adding it later would make them "out of order".

Reply:
I don't think so: we want to be sure that the install strategy is `InstallStrategy.UPDATE_LOCKFILE` to be sure that we get all the best targets that could update the packages.